e2e-byoc-test/agent-client-thin.py

```
@@ -0,0 +1,348 @@
+    for i, step in enumerate(steps):
+        step_lower = step.lower()
+        log(f"Step {i+1}/{len(steps)}: {step}", "AGENT")
+
+        cap_match = re.search(r'using\s+(\S+)', step_lower)
+        cap_hint = cap_match.group(1) if cap_match else ""
+        prompt = re.sub(r'\s*using\s+\S+.*$', '', step, flags=re.IGNORECASE).strip()
+        prompt = re.sub(r'^(create|generate|make)\s+(an?\s+)?', '', prompt, flags=re.IGNORECASE).strip()
+
+        if any(w in step_lower for w in ("video", "animate", "i2v", "t2v", "v2v")):
+            cap = resolve_capability(cap_hint, "video")
+            if "i2v" in cap or "lucy" in cap_lower if "lucy" in step_lower else False:
```

Ruff (0.15.21)

[error] Undefined name `cap_lower`

(F821)